src/Raven.Server/Documents/ETL/Providers/OLAP/OlaptEtl.cs

```diff
@@ -0,0 +1,202 @@
+{
+    public class OlaptEtl : EtlProcess<ToOlapItem, OlapTransformedItems, OlapEtlConfiguration, OlapConnectionString>
+    {
+        public const string OlaptEtlTag = "OLAP ETL";
+
+        public readonly OlapEtlMetricsCountersManager OlapMetrics = new OlapEtlMetricsCountersManager();
+
+        private Timer _timer;
+        private const long MinTimeToWait = 1000;
+        private readonly string _tmpFilePath;
+        private readonly S3Settings _s3Settings;
+
+        public OlaptEtl(Transformation transformation, OlapEtlConfiguration configuration, DocumentDatabase database, ServerStore serverStore)
+            : base(transformation, configuration, database, serverStore, OlaptEtlTag)
+        {
+            Metrics = OlapMetrics;
+
+            var connection = configuration.Connection;
+
+            var localSettings = BackupTask.GetBackupConfigurationFromScript(connection.LocalSettings, x => JsonDeserializationServer.LocalSettings(x),
+                Database, updateServerWideSettingsFunc: null, serverWide: false);
+
+            _s3Settings = BackupTask.GetBackupConfigurationFromScript(connection.S3Settings, x => JsonDeserializationServer.S3Settings(x),
+                    Database, updateServerWideSettingsFunc: null, serverWide: false);
+
+            _tmpFilePath = localSettings?.FolderPath ?? 
+                           (database.Configuration.Storage.TempPath ?? database.Configuration.Core.DataDirectory).FullPath;
+
+            var dueTime = GetDueTime(configuration.RunFrequency);
+
+            _timer = new Timer(_ => _waitForChanges.Set(), null, dueTime, configuration.RunFrequency);
+        }
+
+        private TimeSpan GetDueTime(TimeSpan etlFrequency)
+        {
+            var state = GetProcessState(Database, Configuration.Name, Transformation.Name);
+            if (state.LastBatchTime <= 0) 
+                return TimeSpan.Zero;
+            
+            var nowMs = Database.Time.GetUtcNow().EnsureMilliseconds().Ticks / 10_000;
+            var timeSinceLastBatch = nowMs - state.LastBatchTime;
+
+            var dueTime = etlFrequency.TotalMilliseconds - timeSinceLastBatch;
+
+            if (dueTime < MinTimeToWait)
+                return TimeSpan.Zero;
+
+            return TimeSpan.FromMilliseconds(dueTime);
+        }
+
+        public override EtlType EtlType => EtlType.Olap;
+
+        private static readonly IEnumerator<ToOlapItem> EmptyEnumerator = Enumerable.Empty<ToOlapItem>().GetEnumerator();
+
+        protected override IEnumerator<ToOlapItem> ConvertDocsEnumerator(DocumentsOperationContext context, IEnumerator<Document> docs, string collection)
+        {
+            return new DocumentsToOlapItems(docs, collection);
+        }
+
+        protected override IEnumerator<ToOlapItem> ConvertTombstonesEnumerator(DocumentsOperationContext context, IEnumerator<Tombstone> tombstones, string collection, bool trackAttachments)
+        {
+            return EmptyEnumerator;
```

Comment:
Tombstones are just ignored, right? This is important to write about that explicitly in the documentation